openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/PrepareCore.java

```diff
@@ -268,31 +284,17 @@ void contractNodes() {
             counter++;
             int polledNode = sortedNodes.pollKey();
 
-            //We have worked through all nodes that are not associated with restrictions. Now we can stop the contraction.
-            if (oldPriorities[polledNode] == RESTRICTION_PRIORITY) {
-                //Set the number of core nodes in the storage for use in other places
-                prepareGraph.setCoreNodes(sortedNodes.getSize() + 1);
-                while (!sortedNodes.isEmpty()) {
-                    CHEdgeIterator iter = vehicleAllExplorer.setBaseNode(polledNode);
-                    while (iter.next()) {
-                        if (prepareGraph.getLevel(iter.getAdjNode()) == maxLevel) continue;
-                        prepareGraph.disconnect(vehicleAllTmpExplorer, iter);
-                    }
-                    polledNode = sortedNodes.pollKey();
-                }
-                break;
-            }
-
             if (sortedNodes.getSize() < nodesToAvoidContract) {
                 // skipped nodes are already set to maxLevel
                 prepareGraph.setCoreNodes(sortedNodes.getSize() + 1);
                 //Disconnect all shortcuts that lead out of the core
                 while (!sortedNodes.isEmpty()) {
                     CHEdgeIterator iter = vehicleAllExplorer.setBaseNode(polledNode);
                     while (iter.next()) {
-                        if (prepareGraph.getLevel(iter.getAdjNode()) == maxLevel) continue;
+                        if (prepareGraph.getLevel(iter.getAdjNode()) >= maxLevel) continue;
```

Comment:
`isCoreNode(iter.getAdjNode())`?

Reply:
As far as I can tell this is literally the one and only place where the check for a node being in the core is performed, so I'm not sure it makes sense to abstract it to a function. As mentioned in my previous comment, the fact of a node being at level `< maxLevel` should rather be interpreted in the algorithm context of being contracted rather than not being a core node, so I'm a bit reluctant in recycling such a function for a conceptually different purpose.

Reply:
Understandable. I think from a maintenance perspective, it is still useful to extract a method that has a meaningful name so that someone who reads this can understand what is going on here. Does not have to be `isCoreNode`, just something that explains what this inequality means. But its a minor complaint.